src/pytom_tm/tmjob.py

```diff
@@ -79,6 +80,92 @@ def load_json_to_tmjob(file_name: pathlib.Path, load_for_extraction: bool = True
     return job
 
 
+def _determine_1D_fft_splits(length: int, splits: int, overhang: int = 0) -> list[tuple[tuple[int, int], tuple[int,int]]]:
+        """Split a 1D length into FFT optimal sizes taking into account overhangs
+
+        Parameters
+        ----------
+        length: int
+            Total 1D length to split 
+        splits: int
+            Number of splits to make
+        overhang: int, default 0
+            Minimal overhang/overlap to consider between splits
+
+        Returns
+        -------
+        output: list[tuple[tuple[int, int], tuple[int,int]]]:
+            A list splits where every split gets two tuples meaning:
+              [start, end) of the tomogram data in this split
+              [start, end) of the unique datapoints in this split
+            If a datapoint exists in 2 splits, we add it as unique to 
+            either the split with the most data or the left one if both 
+            splits have the same size
+        """
+        # Everything in this code assumes default slices of [x,y) so including x but excluding y
+        data_slices = []
+        valid_data_slices = []
+        sub_len = []
+        # if single split return early
+        if splits == 1:
+            return [((0, length), (0, length))]
+        # Ceil to guarantee that we map the whole length with enough buffer
+        min_len = int(np.ceil(length / splits)) + overhang
+        min_unique_len = min_len - overhang
+        no_overhang_left = 0
+        while True:
+            if no_overhang_left == 0:
+                # Treat first split specially, only right overhang
+                split_length = next_fast_len(min_len)
+                data_slices.append((0, split_length))
+                valid_data_slices.append((0, split_length-overhang))
+                no_overhang_left = split_length-overhang
+                sub_len.append(split_length)
+            elif no_overhang_left + min_len >= length:
+                # Last slice, only overhang to the left
+                split_length = next_fast_len(min_len)
+                data_slices.append((length-split_length, length))
+                valid_data_slices.append((length-split_length+overhang, length))
+                sub_len.append(split_length)
+                break
+            else:
+                # Any other slice
+                split_length = next_fast_len(min_len+overhang)
```

Comment:
I don't think I fully understand this part. Does `min_len` not already include the overhang? Why is it again added for the central slices?

Reply:
It includes a single overhang (left **or** right), but we want it twice for any center slice (left **and** right)

Reply:
ah! 

I got confused because before I had the overhang as template_size//2. But for the function that does not matter of course! The overhang can be set to whatever.

In that case, could you add back in on lines 467-469, that the input overhang is (template_size//2) + 1 (for uneven shape) ? Due to the overlap from both sides the full region will already be covered.